ush/radmon_err_rpt.sh

```diff
@@ -96,95 +96,95 @@ echo "have_diag_rpt = $have_diag_rpt"
    diag_match=""
    diag_match_len=0 
 
-   if [[ $have_diag_rpt == 1 ]]; then
-      diag_match=$(gawk "/$satname/" $diag_rpt)
+   if [[ ${have_diag_rpt} == 1 ]]; then
+      diag_match=$(gawk "/${satname}/" "${diag_rpt}")
       diag_match_len=$(echo ${#diag_match})
    fi
 
 
-   if [[ $diag_match_len == 0 ]]; then  
+   if [[ ${diag_match_len} == 0 ]]; then
 
-      if [[ $type == "chan" ]]; then
-         echo "looking for match for $satname and $channel"
+      if [[ ${type} == "chan" ]]; then
+         echo "looking for match for ${satname} and ${channel}"
          { while read myline2; do
-            satname2=$(echo $myline2 | gawk '{print $1}')
-            channel2=$(echo $myline2 | gawk '{print $3}')
+            satname2=$(echo "${myline2}" | gawk '{print $1}')
+            channel2=$(echo "${myline2}" | gawk '{print $3}')
 
-            if [[ $satname == $satname2 && $channel == $channel2 ]]; then
-               match="$satname  channel=  $channel" 
-               echo "match from gawk = $match"
+            if [[ ${satname} == ${satname2} && ${channel} == ${channel2} ]]; then
+               match="${satname}  channel=  ${channel}"
+               echo "match from gawk = ${match}"
 	       break;
             else 
 	       match=""
             fi
 
-         done } < $file2
+         done } < "${file2}"
      
 
       else
-         match=$(gawk "/$satname/ && /channel= $channel / && /region= $region /" $file2)
-         echo match = $match
+         match=$(gawk "/${satname}/ && /channel= ${channel} / && /region= ${region} /" "${file2}")
+         echo match = "${match}"
 
          match_len=$(echo ${#match})
-         if [[ $match_len > 0 ]]; then
-            channel2=$(echo $match | gawk '{print $3}')
+         if [[ ${match_len} > 0 ]]; then
```

Comment:
As-is, this doesn't do what was intended.
```suggestion
         if (( match_len > 0 )); then
```